File: 11. maxArea.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxwater = 0
        i, j = 0, len(height)-1
        while i<j:
            water = min(height[i],height[j]) * (j-i)
            if water > maxwater:
                maxwater = water
            if height[i] < height[j]:
                i+=1
            else:
                j-=1
        return maxwater

File: 11. maxArea.java
class Solution {
    public int maxArea(int[] height) {
        int largest = 0;
        int left = 0;
        int right = height.length-1;
        for (int i = 0; i < height.length; i++){
            // System.out.println("i"+i);
            // System.out.println("j"+j);
            int current = Math.min(height[left], height[right])*(right-left);
            largest = Math.max(current, largest);
            if (height[left] > height[right]){
                right--;
            } else {
                left++;
            }
            System.out.println("area" + largest);
        }
        return largest;
    }
}